core/controller/src/main/scala/whisk/core/loadBalancer/LocalLoadBalancerData.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.loadBalancer
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.collection.concurrent.TrieMap
+import scala.concurrent.Future
+import whisk.core.entity.{ActivationId, UUID}
+import whisk.core.entity.InstanceId
+
+/**
+ * Loadbalancer bookkeeping data which are stored locally,
+ * e.g. not shared with other controller instances.
+ *
+ * Note: The state keeping is backed by concurrent data-structures. As such,
+ * concurrent reads can return stale values (especially the counters returned).
+ */
+class LocalLoadBalancerData() extends LoadBalancerData {
+
+  private val activationByInvoker = TrieMap[InstanceId, AtomicInteger]()
+  private val activationByNamespaceId = TrieMap[UUID, AtomicInteger]()
+  private val activationsById = TrieMap[ActivationId, ActivationEntry]()
+  private val totalActivations = new AtomicInteger(0)
+
+  override def totalActivationCount: Future[Int] = Future.successful(totalActivations.get)
```

Comment:
should some of these impls move to the base class?
ie this is the default base class implementation (local)?

Reply:
I thought along the same lines but ended up thinking it might be more confusing to deduplicate 1 or 2 methods vs. having them all implemented in one spot. This impl in particular is not the same for local vs. distributed.